### PR TITLE
Hyperion: enable protobuf/v4l2 and clean up

### PR DESCRIPTION
**Pull request by lrusak**
Thanks @stefansaraev and @Kwiboo 

I've built for RPi and Generic successfully

### CODE REVIEW

File: packages/addons/service/hyperion/package.mk [hunk 1]
```diff
@@ -78,16 +78,23 @@ makeinstall_target() {
 
 addon() {
   mkdir -p $ADDON_BUILD/$PKG_ADDON_ID/bin
-  cp -P $PKG_BUILD/.$TARGET_NAME/bin/hyperiond $ADDON_BUILD/$PKG_ADDON_ID/bin
-  cp -P $PKG_BUILD/.$TARGET_NAME/bin/hyperion-remote $ADDON_BUILD/$PKG_ADDON_ID/bin
+    cp $PKG_BUILD/.$TARGET_NAME/bin/hyperiond $ADDON_BUILD/$PKG_ADDON_ID/bin
+    cp $PKG_BUILD/.$TARGET_NAME/bin/hyperion-remote $ADDON_BUILD/$PKG_ADDON_ID/bin
+    cp $PKG_BUILD/.$TARGET_NAME/bin/hyperion-v4l2 $ADDON_BUILD/$PKG_ADDON_ID/bin
+
+if [ "$KODIPLAYER_DRIVER" = "bcm2835-driver" ]; then
```
Comment (stefansaraev) [inline packages/addons/service/hyperion/package.mk]: spaces

**Comment by lrusak** [comment]
fixed thanks

File: packages/addons/service/hyperion/package.mk [hunk 2]
```diff
@@ -78,16 +78,26 @@ makeinstall_target() {
...
 addon() {
   mkdir -p $ADDON_BUILD/$PKG_ADDON_ID/bin
-  cp -P $PKG_BUILD/.$TARGET_NAME/bin/hyperiond $ADDON_BUILD/$PKG_ADDON_ID/bin
-  cp -P $PKG_BUILD/.$TARGET_NAME/bin/hyperion-remote $ADDON_BUILD/$PKG_ADDON_ID/bin
+    cp $PKG_BUILD/.$TARGET_NAME/bin/hyperiond $ADDON_BUILD/$PKG_ADDON_ID/bin
+    cp $PKG_BUILD/.$TARGET_NAME/bin/hyperion-remote $ADDON_BUILD/$PKG_ADDON_ID/bin
+    cp $PKG_BUILD/.$TARGET_NAME/bin/hyperion-v4l2 $ADDON_BUILD/$PKG_ADDON_ID/bin
+
+  if [ "$KODIPLAYER_DRIVER" = "libamcodec" ]; then
+    cp $PKG_BUILD/.$TARGET_NAME/bin/hyperion-aml $ADDON_BUILD/$PKG_ADDON_ID/bin
```
Comment (stefansaraev) [inline packages/addons/service/hyperion/package.mk]: do you really need this, or is it something required after enabling protobuf support?

Comment (lrusak) [inline packages/addons/service/hyperion/package.mk]: no idea really, it's built when enabling amlogic support.

Comment (stefansaraev) [inline packages/addons/service/hyperion/package.mk]: disregard, those seem to be protobufer remote interface clients. dont hurt. nothing to do with internal amlgrabber. sorry for the noise.

File: packages/addons/addon-depends/protobuf/package.mk [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+PKG_CONFIGURE_OPTS_HOST="--enable-static --disable-shared"
+PKG_CONFIGURE_OPTS_TARGET="--disable-shared --enable-static --with-protoc=$ROOT/$TOOLCHAIN/bin/protoc"
+
+post_makeinstall_target() {
+  rm -rf $INSTALL/usr/bin
+
+  # HACK: we have protoc in $TOOLCHAIN/bin but it seems
+  # the one from sysroot prefix is picked when building hyperion. remove it!
+  rm -f $SYSROOT_PREFIX/usr/bin/protoc
+}
```
Comment (Kwiboo) [inline packages/addons/addon-depends/protobuf/package.mk]: You should only need one of `rm -rf $INSTALL/usr/bin` or `rm -f $SYSROOT_PREFIX/usr/bin/protoc`, the first one should remove `protoc` from `.install_pkg` before it even reaches `$SYSROOT_PREFIX`
Or `rm -f $INSTALL/usr/bin/protoc` to only remove `protoc` from `.install_pkg`

Comment (stefansaraev) [inline packages/addons/addon-depends/protobuf/package.mk]: this all is built as addon. not installed to target image ;)

Comment (Kwiboo) [inline packages/addons/addon-depends/protobuf/package.mk]: ahh, I build my package for target, sorry for noise :)

Comment (lrusak) [inline packages/addons/addon-depends/protobuf/package.mk]: All good, I'm happy to have more eyes looking at things 🚀

**Pull request merged**